File: process_products.js
const fs = require('fs');
const path = require('path');
const https = require('https');

const base_path = 'C:\\Users\\Administrator\\.accio\\accounts\\7087613041\\agents\\DID-F456DA-47F456DAU1779296-1778-553CBB\\project\\jewelry-aura-site';
const json_path = path.join(base_path, 'products.json');
const img_dir = path.join(base_path, 'assets', 'images', 'products');

// Ensure directories exist
if (!fs.existsSync(img_dir)) {
    fs.mkdirSync(img_dir, { recursive: true });
}

// Load products
const data = JSON.parse(fs.readFileSync(json_path, 'utf8'));
const products = data.products || [];

function categorize(title) {
    const t = title.toLowerCase();
    if (t.includes('set') || t.includes('-piece')) return 'Jewelry Sets';
    if (t.includes('necklace') || t.includes('choker') || t.includes('pendant') || t.includes('sweater')) return 'Necklaces';
    if (t.includes('earring') || t.includes('ear dangle') || t.includes('stud')) return 'Earrings';
    if (t.includes('ring')) return 'Rings';
    if (t.includes('bracelet')) return 'Bracelets';
    if (t.includes('bead') || t.includes('material') || t.includes('diy') || t.includes('accessories')) return 'DIY & Materials';
    return 'Fine Jewelry';
}

async function downloadImage(url, dest) {
    return new Promise((resolve, reject) => {
        https.get(url, { headers: { 'User-Agent': 'Mozilla/5.0' } }, (response) => {
            if (response.statusCode !== 200) {
                reject(new Error(`Status Code: ${response.statusCode}`));
                return;
            }
            const file = fs.createWriteStream(dest);
            response.pipe(file);
            file.on('finish', () => {
                file.close();
                resolve();
            });
        }).on('error', (err) => {
            reject(err);
        });
    });
}

async function process() {
    console.log(`Starting processing of ${products.length} products...`);
    
    for (let i = 0; i < products.length; i++) {
        const p = products[i];
        p.category = categorize(p.title);
        
        if (p.image && p.image.startsWith('http')) {
            const filename = `product_${i + 1}.jpg`;
            const dest = path.join(img_dir, filename);
            
            try {
                await downloadImage(p.image, dest);
                p.image = `assets/images/products/${filename}`;
            } catch (err) {
                console.error(`Failed to download ${p.image}: ${err.message}`);
            }
        }
        
        if ((i + 1) % 50 === 0) {
            console.log(`Processed ${i + 1}/${products.length}...`);
        }
    }
    
    fs.writeFileSync(json_path, JSON.stringify(data, null, 2), 'utf8');
    console.log("Processing complete. JSON updated and images saved locally.");
}

process();

File: process_products.py
import json
import os
import urllib.request
import re

# Path config
base_path = r'C:\Users\Administrator\.accio\accounts\7087613041\agents\DID-F456DA-47F456DAU1779296-1778-553CBB\project\jewelry-aura-site'
json_path = os.path.join(base_path, 'products.json')
img_dir = os.path.join(base_path, 'assets', 'images', 'products')

# Ensure directories exist
os.makedirs(img_dir, exist_ok=True)

# Load products
with open(json_path, 'r', encoding='utf-8') as f:
    data = json.load(f)

products = data.get('products', [])

def categorize(title):
    title = title.lower()
    if 'set' in title or '-piece' in title:
        return 'Jewelry Sets'
    if 'necklace' in title or 'choker' in title or 'pendant' in title or 'sweater' in title:
        return 'Necklaces'
    if 'earring' in title or 'ear dangle' in title or 'stud' in title:
        return 'Earrings'
    if 'ring' in title:
        return 'Rings'
    if 'bracelet' in title:
        return 'Bracelets'
    if 'bead' in title or 'material' in title or 'diy' in title or 'accessories' in title:
        return 'DIY & Materials'
    return 'Fine Jewelry'

updated_products = []
total = len(products)

print(f"Starting processing of {total} products...")

for i, p in enumerate(products):
    title = p.get('title', 'Product')
    img_url = p.get('image', '')
    
    # 1. Categorize
    p['category'] = categorize(title)
    
    # 2. Download Image
    if img_url.startswith('http'):
        try:
            # Generate a filename from index or sanitized title
            filename = f"product_{i+1}.jpg"
            save_path = os.path.join(img_dir, filename)
            
            # Simple download
            # We add a header to bypass some basic protections if any
            req = urllib.request.Request(img_url, headers={'User-Agent': 'Mozilla/5.0'})
            with urllib.request.urlopen(req) as response, open(save_path, 'wb') as out_file:
                out_file.write(response.read())
            
            # Update path in JSON (relative to root for frontend)
            p['image'] = f"assets/images/products/{filename}"
        except Exception as e:
            print(f"Failed to download {img_url}: {e}")
    
    updated_products.append(p)
    if (i + 1) % 50 == 0:
        print(f"Processed {i+1}/{total}...")

# Save updated JSON
data['products'] = updated_products
with open(json_path, 'w', encoding='utf-8') as f:
    json.dump(data, f, indent=2, ensure_ascii=False)

print("Processing complete. JSON updated and images saved locally.")
